src/slice.rs

```diff
@@ -143,35 +144,39 @@ impl<'a, T> NonEmptyIterator for NEChunks<'a, T> {
 
     fn next(&mut self) -> Option<Self::Item> {
         if self.index == 0 {
-            let end = self.window.get() - 1;
+            let end = min(self.window.get() - 1, self.tail.len());
 
             let slice = NESlice {
                 head: self.head,
                 tail: &self.tail[0..end],
             };
 
-            self.index = end;
+            // 2024-03-18: This is a workaround for edge case of 1 element slice, thus the index
+            // does not get incremented as end is 0, thus the next iteration will return the same
+            // slice gain and again.
+            self.index = end + 1;
 
             Some(slice)
-        } else if self.index >= self.tail.len() {
+        } else if self.index >= (self.tail.len() + 1) {
             None
         } else {
-            let end = self.index + self.window.get();
-            let slc: &'a [T] = &self.tail[self.index..end];
+            // Ensure we never go out of bounds
+            let end = min(self.index - 1 + self.window.get(), self.tail.len());
+            let slc: &'a [T] = &self.tail[self.index - 1..end];
```

Comment:
Lowering the index by 1 since we had to increase it by 1 to cover the edge case mentioned above.